### ISSUE TITLE
How to Compress Blanks of rectangle selection & How to make custom sort?

### ISSUE THREAD

**Issue by Dean-Corso** [notepad-plus-plus/notepad-plus-plus#13923]
Hey guys,

I'm using Notepad++ and missing some functions or maybe I just don't found them yet and need some advice.
### Description of the Issue

Problem: I'm working with custom rectangle selection using ALT+Drag to select any text parts anywhere and not from the start etc and I wanna trim them all to the first selection start pointer to bring them all starting at same line (remove leading spaces) but this is not working only for entire lines. How to manage that in NP++ when using rectangle selection? I tried other editors and they having such functions like "Notepad3" "Edit > Selection > Compress Blanks    Ctrl+Alt+P" what does work for the rectangle selection. I just did wonder why NP++ doesn't have such a function. Below a small video what I mean and missing in NP++...
![Video_2023-07-23_222959](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/88513086/0604651a-35c9-47e3-96e6-8579492693b1)
...so I just wanna bring all selected lines starting at same line and removing all pre spaces / tabs etc you know.
### Description of the Issue 2

Also have a request about sort functions. I would like to have more possibilities when trying to sort selected lines. Below a example...
```
BBBBBBBBBBBBBBBBBBB
something here
something anywhere
something there

CCCCCCCCCCCCCCCCCCC
something anywhere
something here
something there

AAAAAAAAAAAAAAAAAAA
something here
something there
something anywhere
```
...where you can see three blocks with four lines and I only want to sort the all blocks but only the first line of each block without to touch the three other lines inside of the block. The results at the end should look like this...
```
AAAAAAAAAAAAAAAAAAA
something here
something there
something anywhere

BBBBBBBBBBBBBBBBBBB
something here
something anywhere
something there

CCCCCCCCCCCCCCCCCCC
something anywhere
something here
something there
```
....only the first lines of each block was sorted (ascending) and all other lines of the block was moved. Maybe a little specific but is this also doable anyhow with NP++?

Thank you

**Comment by alankilborn** [comment]
Creating an issue isn't the proper thing to do when you "need advice".
Post [HERE](https://community.notepad-plus-plus.org/).

**Comment by alankilborn** [comment]
If you have this:

```
aaa               111
aaa      222
aaa                               333
aaa                                                   444
```

And you want this:

```
aaa    111
aaa    222
aaa    333
aaa    444
```

Then you can make a "skinny" rectangular selection like this and then press Ctrl+Backspace.

**Comment by alankilborn** [comment]
Your sorting desire is too demanding for Notepad++, but it could be "scripted".  Post in the Community forum with this desire.

**Comment by Dean-Corso** [comment]
Hello @alankilborn,

thank you for the answer. So it works using Strg+Entf keys. Good to know now. But is there also a menu entry what can do that which I didn't seen before? Just see the info in Shortcut mapper / scintilla (SCI_DELWORDLEFT / SCI_DELWORDRIGHT) now.

About my second question of custom sort. Do you have an example script for that? Or do you mean I have to find some PY script and need to install the plugin? So in my case I just want to sort every X line of Y lines block to move with. Could it be doable with any RegEx command? Lets say we take a more simple example like this...
```
BBBBBBBBBBBBBBBBBBB
something here

CCCCCCCCCCCCCCCCCCC
something anywhere

AAAAAAAAAAAAAAAAAAA
something here
```
...and in this case I need to sort every third line started from first line and moving the two lines below for each block. So the distances of all lines would be same.

PS: Not sure about that forum you did mention so the construction of it looks pretty unusual.

Thanks

**Comment by alankilborn** [comment]
> But is there also a menu entry what can do that which I didn't seen before?

No, as you'll notice if you look through the Scintilla commands available in the _Shortcut Mapper_, most do not have menu equivalents.

----------------------------

>custom sort. Do you have an example script for that?

No, I don't, but probably someone on the forum does.

>Could it be doable with any RegEx command?

Again, the forum experts can give a definitive answer on this.
